azurelinuxagent/ga/exthandlers.py

```diff
@@ -1463,6 +1476,11 @@ def _enable_extension(self, extension, uninstall_exit_code):
             # Only save extension state if MC supported
             self.__set_extension_state(extension, ExtensionState.Enabled)
 
+        # start tracking the extension services cgroup.
+        resource_limits = man.get_resource_limits(self.get_full_name(), self.ext_handler.properties.version)
+        CGroupConfigurator.get_instance().start_tracking_extension_services_cgroups(
```

Comment:
we are doing this on each call to enable. The current implementation of CGroupsTelemetry.track_cgroup replaces the current instance of the cgroup if its already been tracked. This loses the state of that instance and should not happen.

Reply:
My understand is that the get_instance() implemented as singleton so we get same instance if it was already instantiated.

Reply:
This is the instance I mean

    @staticmethod
    def track_cgroup(cgroup):
        """
        Adds the given item to the dictionary of tracked cgroups
        """
        if isinstance(cgroup, CpuCgroup):
            # set the current cpu usage
            cgroup.initialize_cpu_usage()

        with CGroupsTelemetry._rlock:
            if not CGroupsTelemetry.is_tracked(cgroup.path):
                CGroupsTelemetry._tracked[cgroup.path] = cgroup <<<<<<<<<<<<<<<<<<<<<<<<<<<<<
                logger.info("Started tracking cgroup {0}", cgroup)

Reply:
If you check above condition we are adding current instance of cgroup if it was not already tracking that cgroup path. So we are not replacing if its already been tracked.

Reply:
ok